commit 46738b1317e169b281ad74690276916e24d1be6d
Author: David Lebrun <david.lebrun@uclouvain.be>
Date:   Tue Nov 15 16:14:04 2016 +0100

    ipv6: sr: add option to control lwtunnel support
    
    This patch adds a new option CONFIG_IPV6_SEG6_LWTUNNEL to enable/disable
    support of encapsulation with the lightweight tunnels. When this option
    is enabled, CONFIG_LWTUNNEL is automatically selected.
    
    Fix commit 6c8702c60b88 ("ipv6: sr: add support for SRH encapsulation and injection with lwtunnels")
    
    Without a proper option to control lwtunnel support for SR-IPv6, if
    CONFIG_LWTUNNEL=n then the IPv6 initialization fails as a consequence
    of seg6_iptunnel_init() failure with EOPNOTSUPP:
    
    NET: Registered protocol family 10
    IPv6: Attempt to unregister permanent protocol 6
    IPv6: Attempt to unregister permanent protocol 136
    IPv6: Attempt to unregister permanent protocol 17
    NET: Unregistered protocol family 10
    
    Tested (compiling, booting, and loading ipv6 module when relevant)
    with possible combinations of CONFIG_IPV6={y,m,n},
    CONFIG_IPV6_SEG6_LWTUNNEL={y,n} and CONFIG_LWTUNNEL={y,n}.
    
    Reported-by: Lorenzo Colitti <lorenzo@google.com>
    Suggested-by: Roopa Prabhu <roopa@cumulusnetworks.com>
    Signed-off-by: David Lebrun <david.lebrun@uclouvain.be>
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/net/ipv6/Makefile b/net/ipv6/Makefile
index 129cad2ba960..a9e9fec387ce 100644
--- a/net/ipv6/Makefile
+++ b/net/ipv6/Makefile
@@ -9,7 +9,7 @@ ipv6-objs :=	af_inet6.o anycast.o ip6_output.o ip6_input.o addrconf.o \
 		route.o ip6_fib.o ipv6_sockglue.o ndisc.o udp.o udplite.o \
 		raw.o icmp.o mcast.o reassembly.o tcp_ipv6.o ping.o \
 		exthdrs.o datagram.o ip6_flowlabel.o inet6_connection_sock.o \
-		udp_offload.o seg6.o seg6_iptunnel.o
+		udp_offload.o seg6.o
 
 ipv6-offload :=	ip6_offload.o tcpv6_offload.o exthdrs_offload.o
 
@@ -23,6 +23,8 @@ ipv6-$(CONFIG_IPV6_MULTIPLE_TABLES) += fib6_rules.o
 ipv6-$(CONFIG_PROC_FS) += proc.o
 ipv6-$(CONFIG_SYN_COOKIES) += syncookies.o
 ipv6-$(CONFIG_NETLABEL) += calipso.o
+ipv6-$(CONFIG_IPV6_SEG6_LWTUNNEL) += seg6_iptunnel.o
+ipv6-$(CONFIG_IPV6_SEG6_HMAC) += seg6_hmac.o
 
 ipv6-objs += $(ipv6-y)
 
@@ -44,7 +46,6 @@ obj-$(CONFIG_IPV6_SIT) += sit.o
 obj-$(CONFIG_IPV6_TUNNEL) += ip6_tunnel.o
 obj-$(CONFIG_IPV6_GRE) += ip6_gre.o
 obj-$(CONFIG_IPV6_FOU) += fou6.o
-obj-$(CONFIG_IPV6_SEG6_HMAC) += seg6_hmac.o
 
 obj-y += addrconf_core.o exthdrs_core.o ip6_checksum.o ip6_icmp.o
 obj-$(CONFIG_INET) += output_core.o protocol.o $(ipv6-offload)